File: src/main/java/com/saybetter/domain/member/application/converter/MemberConverter.java
package com.saybetter.domain.member.application.converter;

import com.saybetter.domain.member.domain.Member;
import com.saybetter.global.common.constant.Provider;
import com.saybetter.global.common.constant.RoleType;

public class MemberConverter {

	private MemberConverter() {
		throw new IllegalStateException("Utility class");
	}

	public static Member toMember(
			final String email,
			final String birthDate,
			final RoleType role,
			final Provider provider,
			final String providerId,
			final String loginId,
			final String name
	) {
		return Member.createMember(email, birthDate, role, provider, providerId, loginId, name);
	}

}

File: core/core-api/src/main/kotlin/io/say/better/domain/member/application/converter/MemberConverter.kt
package io.say.better.domain.member.application.converter

import io.say.better.core.common.constant.Provider
import io.say.better.core.common.constant.RoleType
import io.say.better.storage.mysql.domains.account.entity.Member

class MemberConverter private constructor() {
    init {
        throw IllegalStateException("Utility class")
    }

    companion object {
        fun toMember(
            email: String,
            birthDate: String,
            role: RoleType,
            provider: Provider,
            providerId: String,
            loginId: String,
            name: String,
        ): Member =
            Member.createMember(
                email,
                birthDate,
                role,
                provider,
                providerId,
                loginId,
                name,
            )
    }
}
